code/headers/flame_sensor.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <hwlib.hpp>
+
+namespace r2d2::flame_sensor {
+    class flame_sensor_c {
+    private:
+        hwlib::adc *leds[5];
+        unsigned int flame_threshhold;
+        unsigned int threshhold_error;
+        unsigned int total_detection_angle;
+
+        template <size_t SIZE, class T>
+        size_t array_size(T (&leds)[SIZE]);
```

Comment:
I believe this function can be removed since array size is know via the templated argument SIZE

Reply:
changed https://github.com/R2D2-2019/flame_sensor/pull/1/commits/e71eeeda7b190e0b21024b520ce1826bcaa8f57e

Reply:
Not really what I meant, now the LEDs are hardcoded. however, this is fine for now.